### PR TITLE
Adding custom User-Agent to .NET client

### PR DESCRIPTION
**Pull request by eadgbear**
## 🗣 Description

* Adding a custom user-agent to spice-dotnet to improve the identification of the client when making requests.
* Changing header from `x-spice-user-agent` to standard `User-Agent`

### CODE REVIEW

File: Spice/src/Config/UserAgent.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ namespace Spice.Config;
 
 internal static class UserAgent
 {
-    public static string agent()
+    public static string agent(string? client = null, string? clientVersion = null, string? clientSystem = null, string? clientExtension = null)
```
Comment (sgrebnov) [inline Spice/src/Config/UserAgent.cs:27]: Nit: might be good to make it public so users can benefit and rename it to `SpiceUserAgent`

File: Spice/src/SpiceClientBuilder.cs [hunk 2]
```diff
@@ -80,6 +80,17 @@ public SpiceClientBuilder WithMaxRetries(int maxRetries)
...
     }
 
+    /// <summary>
+    /// Sets the client's user agent.
+    /// </summary>
+    /// <param name="userAgent">User agent string</param>
+    /// <returns>The current instance of <see cref="SpiceClientBuilder"/> for method chaining.</returns>
+    public SpiceClientBuilder WithUserAgent(string userAgent)
+    {
+        _spiceClient.UserAgent = userAgent;
```
Comment (sgrebnov) [inline Spice/src/SpiceClientBuilder.cs:90]: Might be good to check if it is not an empty string

**Review by sgrebnov — DISMISSED** [review]

File: Spice/src/SpiceClientBuilder.cs (hunk 2 above)
Comment (eadgbear) [reply Spice/src/SpiceClientBuilder.cs:90]: We're not doing any validation of the user-agent string provided by the users. It's entirely up to them

File: Spice/src/Config/UserAgent.cs (hunk 1 above)
Comment (eadgbear) [reply Spice/src/Config/UserAgent.cs:27]: Fixed, thanks!

**Review by sgrebnov — APPROVED** [review]

**Pull request merged**